core/src/main/java/com/datastax/oss/driver/api/core/data/CqlVector.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.data;
+
+import com.datastax.oss.driver.shaded.guava.common.base.Joiner;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
+import com.datastax.oss.driver.shaded.guava.common.collect.Iterators;
+import java.util.Arrays;
+
+/** An n-dimensional vector defined in CQL */
+public class CqlVector<T> {
+
+  private final ImmutableList<T> values;
+
+  private CqlVector(ImmutableList<T> values) {
+    this.values = values;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public Iterable<T> getValues() {
+    return values;
```

Comment:
Requiring the use of the iterator pattern here for simple collections types is also making the user write more code that is unnecessary. There should be a simple collection type that can be accessed without all the machinations.